app/src/main/java/vn/ztech/software/projectgutenberg/data/repository/source/repository/book/BookDataSource.kt

```diff
@@ -1,13 +1,35 @@
 package vn.ztech.software.projectgutenberg.data.repository.source.repository.book
 
+import android.content.Context
 import vn.ztech.software.projectgutenberg.data.model.BaseAPIResponse
 import vn.ztech.software.projectgutenberg.data.model.Book
+import vn.ztech.software.projectgutenberg.data.model.BookLocal
+import vn.ztech.software.projectgutenberg.data.model.Resource
 import vn.ztech.software.projectgutenberg.data.repository.OnResultListener
 
 interface BookDataSource {
 
     interface Local {
-        fun getBooksLocal(listener: OnResultListener<List<Book>>)
+        fun getBooksLocal(offset: Int, listener: OnResultListener<List<BookLocal>>)
+        fun scanLocalStorage(
+            context: Context, onResultListener: OnResultListener<Boolean>
+            = object : OnResultListener<Boolean> {
+                    override fun onSuccess(data: Boolean) { // todo leave blank
+                    }
+
+                    override fun onError(e: Exception?) { // todo leave blank
+                    }
+                }
+        )
+
+        fun updateBookImageUrl(res: Resource, fileTitle: String, imgUrl: String)
+        fun deleteBookLocal(
```

Comment:
thêm 1 dòng giữa các function hoặc giữa variable và function nhé a 
```suggestion
        fun updateBookImageUrl(res: Resource, fileTitle: String, imgUrl: String)
        
        fun deleteBookLocal(
```